src/Bicep.Core/Registry/OciModuleRegistry.cs

```diff
@@ -165,34 +153,54 @@ public override async Task<bool> CheckArtifactExists(OciModuleReference referenc
             return mainLayers.Single();
         }
 
+
         private void ValidateModule(OciArtifactResult artifactResult)
         {
             var manifest = artifactResult.Manifest;
-            var artifactType = manifest.ArtifactType;
-
-            // (Bicep v0.20.0 and lower use null for this field, so assume valid in that case
-            if (artifactType is not null &&
-                !allowedArtifactMediaTypes.Contains(artifactType, MediaTypeComparer))
-            {
-                throw new InvalidModuleException(
-                    $"Expected OCI manifest artifactType value of '{BicepMediaTypes.BicepModuleArtifactType}' but found '{artifactType}'. {NewerVersionMightBeRequired}",
-                    InvalidModuleExceptionKind.WrongArtifactType);
-            }
             var config = manifest.Config;
+
             var configMediaType = config.MediaType;
-            if (configMediaType is not null &&
-                !allowedConfigMediaTypes.Contains(configMediaType, MediaTypeComparer))
+            if (configMediaType is not null && !configMediaType.Equals(BicepMediaTypes.BicepModuleConfigV1, MediaTypeComparison))
             {
                 throw new InvalidModuleException($"Did not expect config media type \"{configMediaType}\". {NewerVersionMightBeRequired}");
             }
-
             // Verify nothing wrong with the layers we've been given
             _ = getMainLayer(artifactResult);
 
             // Note: We're not currently writing out non-zero config for modules but expect to soon (https://github.com/Azure/bicep/issues/11482).
             // So ignore the field for now and don't do any validation.
         }
 
+        private static void ValidateProvider(OciArtifactResult artifactResult)
+        {
+            var manifest = artifactResult.Manifest;
+            var config = manifest.Config;
+
+            var configMediaType = config.MediaType;
+            if (configMediaType is not null && !configMediaType.Equals(BicepMediaTypes.BicepProviderConfigV1, MediaTypeComparison))
+            {
+                throw new InvalidModuleException($"Did not expect config media type \"{configMediaType}\".");
+            }
+
+            if (config.Size > 2)
+            {
+                throw new InvalidModuleException("Expected an empty config blob.");
+            }
+
+            if (manifest.Layers.Length != 1)
+            {
+                throw new InvalidModuleException("Expected a single layer in the OCI artifact.");
+            }
+
+            var layer = manifest.Layers.Single();
```

Comment:
Should `ValidateProvider` allow multiple layers (as `ValidateModule` does) so long as exactly one of them has the expected media type? This line will throw an exception unless the manifest has exactly one layer.

Reply:
sounds reasonable, will do.